internal/tagging/aws.go

```diff
@@ -1,14 +1,43 @@
 package tagging
 
-import "github.com/env0/terratag/internal/convert"
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/env0/terratag/internal/convert"
+	"github.com/env0/terratag/internal/tag_keys"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+)
 
 func tagAutoscalingGroup(args TagBlockArgs) (*Result, error) {
-	// for now, we count on it that if there's a single "tag" in the schema (unlike "tags" block),
-	// then no "tags" interpolation is used, but rather multiple instances of a "tag" block
 	// https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html
-	if err := convert.AppendTagBlocks(args.Block, args.Tags); err != nil {
+
+	var tagsMap map[string]string
+	if err := json.Unmarshal([]byte(args.Tags), &tagsMap); err != nil {
 		return nil, err
 	}
 
+	tagsAttr := args.Block.Body().GetAttribute("tags")
+	if tagsAttr != nil {
+		// "tags" interpolation is used
+		tokens := tagsAttr.Expr().BuildTokens(hclwrite.Tokens{})
+		expression := strings.TrimSpace(string(tokens.Bytes()))
+		// may be wrapped with ${ } in TF11
+		expression = strings.TrimPrefix(expression, "${")
+		expression = strings.TrimSuffix(expression, "${")
+
+		key := "local." + tag_keys.GetTerratagAddedKey(args.Filename)
+		newTagsValue := "flatten([" + key + "," + expression + "])"
```

Comment:
Basically, it's going to be something like this:
```
tags = flatten([local.terratag_added_main, flatten(
    [
      {
        key                 = "Name"
        value               = var.cluster_name
        propagate_at_launch = true
      },
      {
        key                 = var.cluster_tag_key
        value               = var.cluster_tag_value
        propagate_at_launch = true
      },
      var.tags,
    ]
  )])

```

Reply:
Guessing the double `flatten` in your example is redundant, right?

Reply:
The original was:
```
flatten(
    [
      {
        key                 = "Name"
        value               = var.cluster_name
        propagate_at_launch = true
      },
      {
        key                 = var.cluster_tag_key
        value               = var.cluster_tag_value
        propagate_at_launch = true
      },
      var.tags,
    ]
  )
```

I this case double flatten is required.